### PR TITLE
feat: feat: generic link effect

### LINKED ISSUE #181: FeedAHelp: Generic Link effects
1. The effect should be visible when the mouse cursor is hovered over the link.
2. The effect should be consistent across all links on the website.
3. The effect should be subtle enough not to distract the user, but still noticeable.
4. The effect should be customizable so that the designer can change the colors, fonts, and other properties to match the overall style of the website.

Source code: https://codepen.io/geoffgraham/pen/RYQybJ

Image

The idea is to make a generic LINK: `<LinkFeedAHelp>Blabla</LinkFeedAHelp>`

### PR DESCRIPTION
**Pull request by sojib-mondol**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #181 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
`GenericLinkEffects.module.css` why module?
where do u see we name css `module`?

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import GenericLinkEffects from './GenericLinkEffects';
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx]: rename this to `GenericLink`

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+test('renders GenericLinkEffects with default props', () => {
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx]: rename this to `GenericLink`

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+
+
+import React, { ReactNode } from 'react';
+import styles from './GenericLinkEffects.module.css';
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx]: WTF why? module?

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+
+
+import React, { ReactNode } from 'react';
+import styles from './GenericLinkEffects.module.css'; 
+
+interface GenericLinkEffectsProps {
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx]: rename

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  children: ReactNode;
+  href: string;
+  color?: string;
+  backgroundColor?: string;
+  fontSize?: string;
+}
+
+const GenericLinkEffects: React.FC<GenericLinkEffectsProps> = ({
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx]: rename

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  children,
+  href,
+  color = 'text-purple-600',
+  backgroundColor = '',
```
Comment (TanvirAlam) [inline packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.tsx]: why do u have emoty css string?

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
rename everything to `GenericLinks`

File: packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx (hunk 1 above)
Comment (sojib-mondol) [reply packages/ui/components/elements/genericLinkEffects/GenericLinkEffects.test.tsx]: resolved

File: apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx [hunk 7]
```diff
@@ -1,9 +1,15 @@
 import React from "react";
+import GenericLink from "../../../../../../../packages/ui/components/elements/GenericLink/GenericLink"
 
 const Ingredients = () => {
   return (
     <div className="mb-2 h-96 w-full border border-red-700">
       This is the Ingredients section.
+
+      <div>
```
Comment (TanvirAlam) [inline apps/feedahelp/src/components/ui/features/ingredients/Ingredients.tsx]: remove this, add the links in the footer

File: packages/ui/components/elements/GenericLink/Blabls.css [hunk 8]
```diff
@@ -0,0 +1,25 @@
+.blabls{
+  border-bottom: 1px solid #453886;
+  color: #453886;
+  padding-bottom: 0.25em;
+  text-decoration: none;
+}
+
+.blabls:hover {
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/Blabls.css]: what fuck is `blabls`? is that bangly word?

File: packages/ui/components/elements/GenericLink/Blabls.css [hunk 9]
```diff
@@ -0,0 +1,25 @@
+.blabls{
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/Blabls.css]: WTF

File: packages/ui/components/elements/GenericLink/index.tsx [hunk 10]
```diff
@@ -0,0 +1 @@
+export * from "./GenericLink";
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/index.tsx]: new line

File: packages/ui/components/elements/GenericLink/Blabls.css [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  text-decoration: none;
+}
+
+.blabls:hover {
+  background-image: url("data:image/svg+xml;charset=utf8,%3Csvg id='squiggle-link' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' xmlns:ev='http://www.w3.org/2001/xml-events' viewBox='0 0 20 4'%3E%3Cstyle type='text/css'%3E.squiggle{animation:shift .3s linear infinite;}@keyframes shift {from {transform:translateX(0);}to {transform:translateX(-20px);}}%3C/style%3E%3Cpath fill='none' stroke='%23453886' stroke-width='2' class='squiggle' d='M0,3.5 c 5,0,5,-3,10,-3 s 5,3,10,3 c 5,0,5,-3,10,-3 s 5,3,10,3'/%3E%3C/svg%3E");
+  background-position: bottom;
+  background-repeat: repeat-x;
+  background-size: 20%;
+  border-bottom: 0;
+  padding-bottom: 0.3em;
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/Blabls.css]: never use `em` for padding

`em` is only used for fonts

File: packages/ui/components/elements/GenericLink/Blabls.css [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+  text-decoration: none;
+}
+
+@keyframes shift {
+  from {
+    transform: translateX(0);
+  }
+  to {
+    transform: translateX(-20px);
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/Blabls.css]: you are translating -20px? r u sure

if so change it to `rem`

File: packages/ui/components/elements/GenericLink/Blabls.styled.tsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+
+
+const getDynamicColor = (color) => {
+  return color;
+};
+
+const BlablsWrapper = styled.a<Props>`
+  border-bottom: 1px solid ${({ color }) => color || '#453886'};
+  color: ${({ color }) => color || '#453886'};
+  padding-bottom: 0.25em;
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/Blabls.styled.tsx]: no `em` only `rem`

File: packages/ui/components/elements/GenericLink/GenericLink.test.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import GenericLink from './GenericLink'; // Assuming the file is named GenericLink.tsx
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/GenericLink.test.tsx]: remove commented code

File: packages/ui/components/elements/GenericLink/GenericLink.test.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('GenericLink component', () => {
+  it('renders correctly with given props', () => {
+    const props = {
+      href: 'https://example.com',
+      color: 'blue',
+      backgroundColor: 'lightgray',
+      fontSize: '16px',
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/GenericLink.test.tsx]: stop testing fontsize, we dont need it, it can change. you test will fail

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
do the changes

File: packages/ui/components/elements/GenericLink/GenericLink.test.tsx (hunk 14 above)
Comment (sojib-mondol) [reply packages/ui/components/elements/GenericLink/GenericLink.test.tsx]: ok

File: packages/ui/components/elements/GenericLink/Blabls.styled.tsx (hunk 13 above)
Comment (sojib-mondol) [reply packages/ui/components/elements/GenericLink/Blabls.styled.tsx]: ok

File: packages/ui/components/elements/GenericLink/Blabls.css (hunk 12 above)
Comment (sojib-mondol) [reply packages/ui/components/elements/GenericLink/Blabls.css]: ok

File: packages/ui/components/elements/GenericLink/GenericLink.styled.tsx [hunk 16]
```diff
@@ -0,0 +1,38 @@
+import styled from 'styled-components';
+type Props = {
+  href:string,
+  color:string,
+  backgroundColor:string,
+  fontSize:string
+}
+
+
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/GenericLink.styled.tsx]: never have mliple empty line, only one.

File: packages/ui/components/elements/GenericLink/GenericLink.styled.tsx [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+const GenericLink = styled.a<Props>`
+  border-bottom: 1px solid ${({ color }) => color || '#453886'};
+  color: ${({ color }) => color || '#453886'};
+  padding-bottom: 0.25rem;
+  text-decoration: none;
+  background-color: ${({ backgroundColor }) => backgroundColor};
+  font-size: ${({ fontSize }) => fontSize || 'initial'};
+  
+
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/GenericLink.styled.tsx]: never have mliple empty line, only one.

File: packages/ui/components/elements/GenericLink/GenericLink.styled.tsx [hunk 18]
```diff
@@ -0,0 +1,38 @@
...
+    background-repeat: repeat-x;
+    background-size: 1rem;
+    border-bottom: none;
+    padding-bottom: 0.3rem;
+    text-decoration: none;
+    
+  }
+`;
+
+
```
Comment (TanvirAlam) [inline packages/ui/components/elements/GenericLink/GenericLink.styled.tsx]: never have mliple empty line, only one.

**Review by TanvirAlam — CHANGES_REQUESTED** [review]
so far so good, do the smale changes we will merge it

**Review by TanvirAlam — APPROVED** [review]
LGTM so far

**Pull request merged**